spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/validators/PoliciesValidator.java

```diff
@@ -60,11 +62,19 @@ private boolean isValidObject(Object o) {
     }
 
     private boolean isValidPolicy(Policy policy) {
-        return (policy.getLimit() != null || policy.getQuota() != null) && isValidRoles(policy);
+        return (policy.getLimit() != null || policy.getQuota() != null) && isValidRoles(policy) && isValidUrlPattern(policy);
     }
 
     private boolean isValidRoles(Policy policy) {
+        return isValid(policy, ROLE);
+    }
+
+    private boolean isValidUrlPattern(Policy policy) {
+        return isValid(policy, URL_PATTERN);
+    }
+
+    private boolean isValid(Policy policy, RateLimitType rateLimitType) {
         return policy.getType().stream()
-                .noneMatch(type -> type.getType().equals(ROLE) && type.getMatcher() == null);
```

Comment:
Maybe it is better to add `isValid` method to `RateLimitType` and run it on all the policy types